tests/agentzh.t

```diff
@@ -113,7 +113,78 @@ print(cjson.encode(data))
 
 
 
-=== TEST 9: multiple calls to lua_cjson_new (1/2)
+=== TEST 9: array_mt on empty tables
+--- lua
+local cjson = require "cjson"
+local data = {}
+setmetatable(data, cjson.array_mt)
+print(cjson.encode(data))
+--- out
+[]
+
+
+
+=== TEST 10: array_mt on non-empty tables
+--- lua
+local cjson = require "cjson"
+local data = { "foo", "bar" }
+setmetatable(data, cjson.array_mt)
+print(cjson.encode(data))
+--- out
+["foo","bar"]
+
+
+
+=== TEST 11: array_mt on non-empty tables with holes
+--- lua
+local cjson = require "cjson"
+local data = {}
+data[1] = "foo"
+data[2] = "bar"
+data[4] = "last"
+data[9] = "none"
+setmetatable(data, cjson.array_mt)
+print(cjson.encode(data))
+--- out
+["foo","bar",null,"last"]
+
+
+
+=== TEST 12: array_mt on tables with hash part
+--- lua
+local cjson = require "cjson"
+local data
+
+if jit and string.find(jit.version, "LuaJIT 2.1.0", nil, true) then
```

Comment:
Here, we follow this approach instead of a safe `require` with `pcall` to fail hard in case we are testing against a version of LuaJIT that does not have `table.new`.